runelite-client/src/main/java/net/runelite/client/plugins/customcursor/CustomCursorPlugin.java

```diff
@@ -110,8 +129,36 @@ private void updateCursor()
 			}
 			return;
 		}
+		else if (selectedCursor == CustomCursor.EQUIPPED_WEAPON)
+		{
+			final ItemContainer itemContainer = client.getItemContainer(InventoryID.EQUIPMENT);
+
+			if (itemContainer == null)
+			{
+				clientUI.resetCursor();
+				return;
+			}
+
+			final Item weapon = itemContainer.getItems()[EquipmentInventorySlot.WEAPON.getSlotIdx()];
+			final BufferedImage image = getImage(weapon);
+
+			if (weapon.getQuantity() > 0)
+			{
+				clientUI.setCursor(image, selectedCursor.getName());
+			}
+			else
+			{
+				clientUI.resetCursor();
+			}
+			return;
+		}
 
 		assert selectedCursor.getCursorImage() != null;
 		clientUI.setCursor(selectedCursor.getCursorImage(), selectedCursor.getName());
 	}
+
+	private BufferedImage getImage(Item item)
+	{
+		return itemManager.getImage(item.getId(), item.getQuantity(), item.getQuantity() > 1);
```

Comment:
What weapons have different sprites for high quantities? [All the ones listed on the wiki](https://oldschool.runescape.wiki/w/Weapons#Thrown_weapons) have identical sprites regardless of quantity.